Share/NCShareExtension.swift

```diff
@@ -177,17 +177,22 @@ class NCShareExtension: UIViewController {
         collectionView.reloadData()
         tableView.reloadData()
     }
+    
+    override func viewWillDisappear(_ animated: Bool) {
+        super.viewWillDisappear(animated)
+        self.cancel(with: .cancel)
+    }
 
```

Comment:
Please test it: if the user dismiss the share without the cancel button (to drag down) the current upload metadata is correctly removed

Reply:
I did have some issues with calling cancel multiple times before, even causing runtime crashes..
But I think the main problem is this wired behaviour when trying to dismiss:

![ScreenRecording](https://user-images.githubusercontent.com/18512366/150339051-f06a8430-16b0-4788-a019-fab7c3186ba7.mov)